couchdb/couchdb_startup.sh

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env bash
```

Comment:
I used this script because it allows us to run the server and signer with less privileged users. The less privileged users cannot create a database themselves but can use a database setup by an admin. I get the following type of an error otherwise. So, yeah, its scripted bootstrapping of the database tables for the purpose of privilege separation.
```
{"level":"fatal","msg":"unable to create DB notarysigner$private_keys: Unauthorized: You are not a server admin.","time":"2018-05-21T15:02:13Z"}
```

The privilege separation was mainly for CouchDB. For Cloudant, iirc, we cannot setup other users with their own credentials but have to use the account given to use by the cloud.

Reply:
I added the reason for this script to the top of the file including documentation regarding the environment variables that are needed.

Reply:
In https://github.com/theupdateframework/notary/pull/1279#discussion_r195611624 I mentioned that maybe the admin connection and the user connection should be the same.  I'm going to amend that by saying that maybe when establishing the admin connection for bootstrapping (which can call the same function), maybe we can check for the admin user/passphrase environment variable?

If I am understanding everything correctly, currently the URL that is passed means that all connections are made with the server/signer user and passphrase, which means that you can't establish an admin connection.  But if you got the username and passphrase differently if `bootstrap` is set to true, then when doing the bootstrap you would be able to connect as an administrator and generate the tables, etc. via the code that has already been written.

Reply:
The environment variable to get the admin user/passphrase from is DBURL ?

If we go through the bootstrap we would need a way to pass two usernames (signer and server) for the two databases along with two passwords. How do we do that?

Reply:
MySQL/PostreSQL are initialized using SQL scripts in `notarysql/mysql-initdb.d/initial-notaryserver.sql` for example. There the notaryserver database is created as well as the server user. The server user is granted access to the notaryserver database. 
```
CREATE DATABASE IF NOT EXISTS `notaryserver`;

CREATE USER "server"@"%" IDENTIFIED BY "";

GRANT
        ALL PRIVILEGES ON `notaryserver`.* 
        TO "server"@"%";
```

The SQL files are run by the MySQL database container where these files are mapped into the container's `docker-entrypoint-initdb.d` directory. 

So I would justify this script here to be an equivalent of the MySQL bootstrapping with a couple of differences:
- there's no built-in bootstrapping mechanism in the CouchDB container that would run some setup scripts like those SQL scripts mapped into a certain directory; so we manipulate the `entrypoint` and `command` of the CouchDB container
- we cannot create databases in CouchDB but have to create tables instead and grant access to these tables. 
- we set the passwords for the signer and server users here so we don't have to provide the admin password to the server and signer containers for the bootstrapping.
- CouchDB needs a couple of non-Notary databases setup during bootstrapping: `_users`, `_replicator`, `_global_changes`. These are described here: http://docs.couchdb.org/en/latest/install/setup.html